templates/e-commerce/product-details.amp.html

```diff
@@ -0,0 +1,183 @@
+<!---
+Copyright 2017 The AMP Start Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<!doctype html>
+<html ⚡>
+{{#header}} {{> ../../utils/amp-page-head.snip.html}} {{/header}}
+
+<body>
+  {{#header}}{{> ../../components/navbar/navbar.snip.html}} {{/header}} {{#main}}
+  <main id="content" role="main" class="main">
+    {{#product-details}}
+    <amp-state id="product">
+      <script type="application/json">
+        {
+        "price": "{{price}}",
+        "selectedColor": "white",
+        {{#colours}}
+          "{{name}}": {
+            "thumb": {
+              "image1": "{{{thumb.image1}}}",
+              "image2": "{{{thumb.image2}}}",
+              "image3": "{{{thumb.image3}}}"
+            },
+            "large": {
+              "image1": "{{{large.image1}}}",
+              "image2": "{{{large.image2}}}",
+              "image3": "{{{large.image3}}}"
+            }
+          },
+        {{/colours}}
+        "relatedProductsSrc": "/json/related_products.json",
+        "selectedSlide": 0
+      }
+      </script>
+    </amp-state>
+
+    <section class="flex flex-wrap">
+      <div class="col-12 md-col-6 px2 md-pl7 md-pt5 pt2">
+        <amp-carousel width="1280" height="720" layout="responsive" type="slides" [slide]="product.selectedSlide" on="slideChange: AMP.setState({product: {selectedSlide: event.index}})">
+          <amp-img [src]="product[product.selectedColor].large.image1" src="{{colours.0.large.image1}}" width="1280" height="720" layout="responsive"
+            role="button" tabindex="0" alt="product image"></amp-img>
+          <amp-img [src]="product[product.selectedColor].large.image2" src="{{colours.0.large.image2}}" width="1280" height="720" layout="responsive"
+            role="button" tabindex="0" alt="product image"></amp-img>
+          <amp-img [src]="product[product.selectedColor].large.image3" src="{{colours.0.large.image3}}" width="1280" height="720" layout="responsive"
+            role="button" tabindex="0" alt="product image"></amp-img>
+        </amp-carousel>
+
+        <amp-selector class="center" [selected]="product.selectedSlide" on="select:AMP.setState({product: {selectedSlide: event.targetOption}})">
+          <ul class="list-reset inline-block">
+            <li class="inline-block commerce-product-thumb">
+              <amp-img option="0" layout="responsive" selected="selected" [src]="product[product.selectedColor].thumb.image1" src="{{colours.0.thumb.image1}}"
```

Comment:
I didn't create a loop for images here since it's not possible in mustache to know when the first iteration of the loop is taking place without custom functions.